src/Symfony/Component/Debug/FatalErrorHandler/UndefinedMethodFatalErrorHandler.php

```diff
@@ -36,8 +36,12 @@ public function handleError(array $error, FatalErrorException $exception)
 
         $message = sprintf('Attempted to call an undefined method named "%s" of class "%s".', $methodName, $className);
 
+        if (null === $methods = get_class_methods($className)) {
+            return new UndefinedMethodException($message, $exception);
+        }
+
         $candidates = array();
-        foreach (get_class_methods($className) as $definedMethodName) {
+        foreach ($methods as $definedMethodName) {
```

Comment:
Casting to an array is something commonly used in the Symfony codebase, even in more critical paths where micro optimizations would matter more. To me the intention is clear, but I hear your arguments.

Also the issue here is more about `class@anonymous` not being a class name, which is why `get_class_methods` returns `null` as a way to notify there is an issue with the function call, not as I first thought an inconsistency due to an anonymous class without any method. You don't access the genearted internal class name here nor the instance, so it's expected to get `null`.